src/loaders/storage/loadRlLoadout.ts

```diff
@@ -0,0 +1,63 @@
+import { createPaintConfig, RocketAssetManager, BodyModel, WheelsModel, Body, Wheel, Decal } from 'rl-loadout-lib'
+import { ExtendedPlayer } from '../../models/ReplayMetadata'
+import { Mesh, MeshPhongMaterial, MeshStandardMaterial } from 'three'
+
+export async function loadRlLoadout(manager: RocketAssetManager, player: ExtendedPlayer, defaultLoadout?: boolean):
+  Promise<{ body: BodyModel; player: ExtendedPlayer }> {
+  let body: BodyModel
+  let wheels: WheelsModel
+
+  if (defaultLoadout) {
+    const paintConfig = createPaintConfig(player.isOrange)
+    body = new BodyModel(Body.DEFAULT, Decal.NONE, paintConfig, manager.config)
+    wheels = new WheelsModel(Wheel.DEFAULT, paintConfig, manager.config)
+    await Promise.all([body.load(), wheels.load()])
```

Comment:
Does your library handle caching the body and wheel loads?

Reply:
The lib itself doesn't handle caching, but I've enabled three's internal cache here: https://github.com/SaltieRL/WebReplayViewer/blob/6b0163d1bc843609b8f6876c21fd1e4963c9aa7c/src/builders/SceneBuilder.ts#L29

Although since this only caches the raw downloaded assets, they do still need to be processed. If you want to speed up processing, there are also smaller textures available that have 4x less pixels.

Reply:
Three's cache is fine for handling preexisting assets being added to a scene, but this might still make the request to download these models N times over assuming the browser isn't smart about cache. On a mobile connection, another 30Mb is another 30 seconds.